### PR TITLE
Added all details about logic module 

### PR DESCRIPTION
**Pull request by sumith1896**
I have included all the information about our `Logic` module in the txt file here for feedback on what has to be included and what not. AFAIK this is self contained, information has been collected from docs, code and references given in code. 

Once we decide on that, I'll make a fresh PR with the info added to `tex` with suitable examples and appropriate citations.

/cc @asmeurer @certik @moorepants @scopatz

### CODE REVIEW

**Comment by sumith1896** [comment]
@asmeurer  Could you have a look at this too?

File: sympy-paper-logic.txt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+constructing corresponding expressions.
+
+These boolean symbols can be substituted `True` or `False` 
+using `.subs()`
+
+Example:
+>>> (y & x).subs({x: True, y: True})
+True
+
+SOPform and POSform
```
Comment (asmeurer) [inline sympy-paper-logic.txt]: I wouldn't mention this.

File: sympy-paper-logic.txt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+Algorithms for SAT solving
+--------------------------
+`dpll`
+------
+Has basic DPLL implementation with DP kernalization, branching
+and backtracking.
+- http://en.wikipedia.org/wiki/DPLL_algorithm
+- https://www.researchgate.net/publication/242384772_Implementations_of_the_DPLL_Algorithm
+
+`dpll2`
```
Comment (asmeurer) [inline sympy-paper-logic.txt]: I would only mention this, as "the DPLL algorithm"

Comment (asmeurer) [inline sympy-paper-logic.txt]: Even then, the DPLL algorithm is a classic algorithm. I don't think it's worth saying much more than "SymPy has an implementation of the DPLL algorithm" (with a reference).

File: sympy-paper-logic.txt [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+Parameters:
+
+expr: The expression to check for satisfiability.
+algorithm: Internal algorithm to be used. Current options in 
+	SymPy are 'dpll' and 'dpll2', to be explained in following
+	sections.
+all_models: On setting all_models to True, if given expr is 
+	satisfiable then it returns a generator of models. However, 
+	if expr is unsatisfiable then it returns a generator containing 
+	the single element False.
```
Comment (asmeurer) [inline sympy-paper-logic.txt]: This isn't API documentation. You should describe `satisfiable` in a high-level way, and give an example.

**Comment by asmeurer** [comment]
Please write this up in LaTeX format and include it in the paper.

File: features_logic.tex [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+SymPy supports construction and manipulation of boolean expressions
+through the \texttt{logic} module. SymPy symbols can be used as 
+propositional variables and also be substituted as \texttt{True} 
+or \texttt{False}. A good number of manipulation features for boolean 
+expressions have been implemented in the \texttt{logic} module.
+
+\subsubsection{Constructing boolean expressions}
+
+A boolean variable can be declared as a SymPy symbol. Python
+operators \&, \textbar  and \textasciitilde are overloaded for logical \texttt{And},
```
Comment (leosartaj) [inline features_logic.tex]: \textbar{}

Comment (leosartaj) [inline features_logic.tex]: \textasciitilde{}

File: features_logic.tex [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+propositional variables and also be substituted as \texttt{True} 
+or \texttt{False}. A good number of manipulation features for boolean 
+expressions have been implemented in the \texttt{logic} module.
+
+\subsubsection{Constructing boolean expressions}
+
+A boolean variable can be declared as a SymPy symbol. Python
+operators \&, \textbar  and \textasciitilde are overloaded for logical \texttt{And}, 
+\texttt{Or} and \texttt{negate}. Several others like \texttt{Xor},
+\texttt{Implies} can be constructed with \^{}, \textgreater\textgreater respectively.
```
Comment (leosartaj) [inline features_logic.tex]: \textgreater\textgreater{}

File: features_logic.tex [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+operators \&, \textbar  and \textasciitilde are overloaded for logical \texttt{And}, 
+\texttt{Or} and \texttt{negate}. Several others like \texttt{Xor},
+\texttt{Implies} can be constructed with \^{}, \textgreater\textgreater respectively.  
+The above are just a shorthand, expressions can also be constructed
+by directly calling \texttt{And()}, \texttt{Or()}, \texttt{Not()},
+\texttt{Xor()}, \texttt{Nand()}, \texttt{Nor()}, etc.
+The boolean symbols can also be substituted \texttt{True} or \texttt{False}
+
+\begin{verbatim}
+>>> e = (x & y) | z
```
Comment (leosartaj) [inline features_logic.tex:21]: IMO, you should import from sympy and create the necessary symbols and stuff. That way somebody following along can just copy and paste the examples in say an IPython shell.

**Comment by leosartaj** [comment]
The examples have `x, y, z, A, B, C, a, b, c`. It can be difficult to follow along. This makes it harder to just copy paste and run the examples.

File: features_logic.tex [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+\subsubsection{Constructing boolean expressions}
+
+A boolean variable can be declared as a SymPy symbol. Python
+operators \&, \textbar  and \textasciitilde are overloaded for logical \texttt{And}, 
+\texttt{Or} and \texttt{negate}. Several others like \texttt{Xor},
+\texttt{Implies} can be constructed with \^{}, \textgreater\textgreater respectively.  
+The above are just a shorthand, expressions can also be constructed
+by directly calling \texttt{And()}, \texttt{Or()}, \texttt{Not()},
+\texttt{Xor()}, \texttt{Nand()}, \texttt{Nor()}, etc.
+The boolean symbols can also be substituted \texttt{True} or \texttt{False}
```
Comment (leosartaj) [inline features_logic.tex]: This line is similar to Line-5

**Comment by moorepants** [comment]
@Sumith1896 Can you please address the comments here so that we can merge today?

**Comment by sumith1896** [comment]
There seems to be extra references in the bib file. Maybe I fixed the merge conflicts the wrong way. Should I remove those references and keep only what I have used?

**Comment by sumith1896** [comment]
@moorepants @certik ping

**Comment by certik** [comment]
@Sumith1896 I don't understand your question. Are you saying your PR added references that we don't need? I would remove them then.

**Comment by sumith1896** [comment]
Cool, I think it is good to go for the first draft. I'll update the content accordingly as the paper evolves.

**Comment by certik** [comment]
@moorepants btw, we should add an error on unresolved references, as we currently have some.

**Comment by sumith1896** [comment]
Thanks @certik

**Pull request merged**